### PR TITLE
Feature/withdraw aave

### PR DESCRIPTION
**Pull request by ialberquilla**
## Description

Implementes withdraw condition to allow delegator withdraw collateral + fees

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [x] Tests Cover Changes
- [ ] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

File: contracts/conditions/defi/aave/AaveCollateralWithdrawCondition.sol [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import '../../Condition.sol';
+import '../../../registry/DIDRegistry.sol';
+import './AaveCreditVault.sol';
+import '../../../Common.sol';
+import '../../../templates/AaveCreditTemplate.sol';
+import '@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol';
+
+/**
+ * @title Aave Collateral Deposit Condition
```
Comment (aaitor) [inline contracts/conditions/defi/aave/AaveCollateralWithdrawCondition.sol:15]: Update documentation

File: contracts/conditions/defi/aave/AaveCollateralWithdrawCondition.sol [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    address _collateralAsset
+  )
+    external
+    payable
+    nonReentrant
+    returns (ConditionStoreLibrary.ConditionState)
+  {
+    // Withdraw the collateral from the Aave Lending pool contract and the agreement fees
+    AaveCreditVault vault = AaveCreditVault(_vaultAddress);
+    vault.withdrawCollateral(_collateralAsset, _delegator);
```
Comment (aaitor) [inline contracts/conditions/defi/aave/AaveCollateralWithdrawCondition.sol:87]: Do we need to check if a previous condition is fulfilled?

File: contracts/conditions/defi/aave/AaveCreditVault.sol [hunk 3]
```diff
@@ -27,13 +27,23 @@ contract AaveCreditVault is
...
   uint256 private constant FEE_BASE = 10000;
+  address private treasuryAddress;
 
+  /**
+   * Vault constructor, creates a unique vault for each agreement
+   * @param _lendingPool Aave lending pool address
+   * @param _dataProvider Aave data provider address
+   * @param _weth WETH address
+   * @param _nvmFee Nevermined fee that will apply to this agreeement
+   * @param _agreementFee Agreement fee that lender will receive on agreement maturity
```
Comment (aaitor) [inline contracts/conditions/defi/aave/AaveCreditVault.sol:38]: missing documentation of `_agreementFee` and `_treasuryAddress`

File: contracts/conditions/defi/aave/AaveCollateralWithdrawCondition.sol (hunk 2 above)
Comment (ialberquilla) [reply contracts/conditions/defi/aave/AaveCollateralWithdrawCondition.sol:87]: Yes, the repayment condition should be fulfilled, how can I check that?

Comment (ialberquilla) [reply contracts/conditions/defi/aave/AaveCollateralWithdrawCondition.sol:87]: @aaitor If you approve this PR I'll create a new one only with the validations of the previous conditions.

**Pull request merged**